vllm/reasoning/ernie45_reasoning_parser.py

```diff
@@ -0,0 +1,207 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright contributors to the vLLM project
+
+from collections.abc import Sequence
+from typing import Optional, Union
+
+from transformers import PreTrainedTokenizerBase
+
+from vllm.entrypoints.openai.protocol import (ChatCompletionRequest,
+                                              DeltaMessage)
+from vllm.logger import init_logger
+from vllm.reasoning import ReasoningParser, ReasoningParserManager
+
+logger = init_logger(__name__)
+
+
+@ReasoningParserManager.register_module("ernie45")
+class Ernie45ReasoningParser(ReasoningParser):
+    """
+    Reasoning parser for Ernie45 thinking model.
+    The Ernie45 thinking model ouput format is
+        abc\n</think>\n\n\n<response>\ndef\n</response>\n
+    or  abc\n</think>\ndef
+    or  abc\n</think>\n\n\n<tool_call>\nxyz\n</tool_call>\n
+    """
+
+    think_start_token_id: int
+    think_end_token_id: int
+    response_start_token_id: int
+    response_end_token_id: int
+    tool_call_start_token_id: int
+    tool_call_end_token_id: int
+
+    think_start_token: str = "<think>"
```

Comment:
Could you take a look at whether the current implementation can be simplified based on `BaseThinkingReasoningParser`?

Reply:
Yes, tool_call related tokens can be deleted